packages/@lwc/rollup-plugin/src/index.ts

```diff
@@ -30,7 +30,11 @@ export interface RollupLwcOptions {
     /** The configuration to pass to the `@lwc/template-compiler`. */
     preserveHtmlComments?: boolean;
     /** The configuration to pass to `@lwc/compiler`. */
-    experimentalDynamicComponent?: DynamicComponentConfig;
+    experimentalDynamicComponent?: DynamicImportConfig;
```

Comment:
IMO, it's quite tedious to update the `RollupLwcOptions` interface every time a new flag is added. It might be interesting to define the `RollupLwcOptions` interface as a union with the LWC compiler config interface.